From 7df886cd852f0fa3a992cb85049971f363a37ec5 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Wed, 19 Nov 2014 14:24:16 +0800
Subject: [PATCH 0675/1691] MLK-11488-8 arm: imx: adjust qspi device index
 according to dtb setting

commit 7df886cd852f0fa3a992cb85049971f363a37ec5 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

When resume from DSM with Mega/Fast off, we need to restore
the right QSPI module for M4, so get the qspi index from dtb file.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/pm-imx6.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/pm-imx6.c b/arch/arm/mach-imx/pm-imx6.c
index 7a49eca..4253ac2 100644
--- a/arch/arm/mach-imx/pm-imx6.c
+++ b/arch/arm/mach-imx/pm-imx6.c
@@ -1150,7 +1150,8 @@ void __init imx6sx_pm_init(void)
 	if (np)
 		console_base = of_iomap(np, 0);
 	if (imx_src_is_m4_enabled()) {
-		np = of_find_node_by_path("/soc/aips-bus@02100000/qspi@021e4000");
+		np = of_find_compatible_node(NULL, NULL,
+				"fsl,imx6sx-qspi-m4-restore");
 		if (np)
 			qspi_base = of_iomap(np, 0);
 		WARN_ON(!qspi_base);
-- 
1.9.1

